compiler/rustc_trait_selection/src/traits/query/dropck_outlives.rs

```diff
@@ -350,7 +350,7 @@ pub fn dtorck_constraint_for_ty_inner<'tcx>(
             // Note that we don't care about whether the resume type has any drops since this is
             // redundant; there is no storage for the resume type, so if it is actually stored
             // in the interior, we'll already detect the need for a drop by checking the interior.
-            let typing_env = tcx.erase_regions(typing_env);
+            let typing_env = tcx.erase_and_anonymize_regions(typing_env);
```

Comment:
existing, but that seems very off. Erasing regions in the typing_env is not rly something we do anywhere else.

can you add
```suggestion
            // FIXME(@lcnr): Why do we erase regions in the env here? Seems odd
            let typing_env = tcx.erase_and_anonymize_regions(typing_env);
```